core/workspace.js

```diff
@@ -490,8 +491,8 @@ Blockly.Workspace.prototype.fireChangeListener = function(event) {
  */
 Blockly.Workspace.prototype.getBlockById = function(id) {
   var block = this.blockDB_[id];
-  if (!block && this.getFlyout() && this.getFlyout().getWorkspace()) {
-    block = this.getFlyout().getWorkspace().blockDB_[id];
+  if (!block && this.getFlyout_() && this.getFlyout_().getWorkspace()) {
```

Comment:
If you're using `getFlyout` in scratch-gui, you shouldn't rename it to `getFlyout_`, because that means it should be private.